kvstore.sh

```diff
@@ -69,14 +56,25 @@ _lock_then () {
     echo "error: nothing to lock" &>2
     return 1
   fi
-  if ! _lockfile "$lockfile"; then
-    echo "error: could not aquire lock" &>2
-    return 2
+  if type flock &>/dev/null; then
+    set -e
+    (
+      flock -w 5 -x 200
+      "$cmd" "$@"
+    ) 200>"$lockfile"
+    set +e
```

Comment:
I actually hadn't followed the link, but I had read that same doc earlier this week for an unrelated problem 😆